src/passkey_child/passkey_child_common.c

```diff
@@ -319,3 +341,48 @@ register_key(struct passkey_data *data)
 
     return ret;
 }
+
+errno_t
+format_pem_public_key(TALLOC_CTX *mem_ctx, struct passkey_data *data,
+                      char **_pem_key)
+{
+    const unsigned char *public_key = NULL;
+    size_t pk_length;
+    errno_t ret;
+
+    public_key = sss_base64_decode(mem_ctx, data->public_key, &pk_length);
+    if (public_key == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "sss_base64_decode failed.\n");
+        ret = EINVAL;
+        goto done;
+    }
+
+    switch (data->type) {
+    case COSE_ES256:
+        ret = es256_pubkey_to_pem(mem_ctx, public_key, pk_length, _pem_key);
+        if (ret != EOK) {
+            goto done;
+        }
+        break;
+    case COSE_RS256:
+        ret = rs256_pubkey_to_pem(mem_ctx, public_key, pk_length, _pem_key);
+        if (ret != EOK) {
+            goto done;
+        }
+        break;
+    case COSE_EDDSA:
+        ret = eddsa_pubkey_to_pem(mem_ctx, public_key, pk_length, _pem_key);
+        if (ret != EOK) {
+            goto done;
+        }
+        break;
+    default:
+        DEBUG(SSSDBG_OP_FAILURE, "Invalid key type.\n");
```

Comment:
Is it really an intent to return `EOK` in this case?

Reply:
Fixed.